src/Raven.Server/Rachis/CandidateAmbassador.cs

```diff
@@ -49,20 +51,23 @@ public void Start()
         public void Dispose()
         {
             _disposed = true;
-            _conenctToPeer?.Dispose();
-            if (_thread != null && _thread.ManagedThreadId != Thread.CurrentThread.ManagedThreadId)
+            if (ElectionWon == false)
```

Comment:
I don't like this, there is a race condition here between the disposal and the other candidate getting the value.
Better to make it explicit ownership change. First, the candidate take ownership on the connections, and in thus doing set the connection to null.